### ISSUE TITLE
panics if run in package main

### ISSUE THREAD

**Issue by tv42** [vektah/gqlgen#170]
gqlgen really doesn't seem to like package main, but even more mystifying is how it comes up with the package name "example". And obviously command-line utilities shouldn't panic on bad input.

```
tv@dark ~/z$ tree
.
└── src
    └── example.com
        └── foo-bar
            ├── main.go
            ├── schema.graphql
            └── types.json

3 directories, 3 files
tv@dark ~/z$ cd src/example.com/foo-bar/
tv@dark ~/z/src/example.com/foo-bar$ GOPATH="$HOME/z:$(go env GOPATH)" go generate
panic: cannot find package "example" in any of:
	/home/tv/src/go-1.10/src/example (from $GOROOT)
	/home/tv/z/src/example (from $GOPATH)
	/home/tv/go/src/example

goroutine 1 [running]:
github.com/vektah/gqlgen/codegen.(*Imports).add(0xc427326540, 0xc420138be0, 0x7, 0xc4273c73e0)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/import_build.go:61 +0x2a3
github.com/vektah/gqlgen/codegen.buildImports(0xc420070e10, 0xc42013e2d0, 0x22, 0xc420062e80)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/import_build.go:40 +0x194
github.com/vektah/gqlgen/codegen.(*Config).models(0xc4201f1dc8, 0x30, 0x732840, 0x897228)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/build.go:40 +0xab
github.com/vektah/gqlgen/codegen.Generate(0xc42013e270, 0x24, 0xc420070cc0, 0xc4200ab420, 0xc42013e330, 0x2f, 0xc4201137b8, 0x4, 0xc420138c2f, 0x10, ...)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/codegen.go:45 +0xa0
main.main()
	/home/tv/go/src/github.com/vektah/gqlgen/main.go:42 +0x1e7
main.go:3: running "gqlgen": exit status 2
[EXIT:1] tv@dark ~/z/src/example.com/foo-bar$ head -99 *
==> main.go <==
package main

//go:generate gqlgen -package main -modelpackage main -typemap types.json -schema schema.graphql

func main() {
}

==> schema.graphql <==
type Greeting {
  Message: String
}

==> types.json <==
{
  "Greeting": "example.com/foo-bar"
}
tv@dark ~/z/src/example.com/foo-bar$ 
```

**Comment by tv42** [comment]
Ah yes, that caused that confusion. I'm still seeing some funky stuff, I'll either provide more detail here or close the bug soon. Thanks.

**Comment by tv42** [comment]
Okay, my remaining confusion is completely about the two generated files, and the directories they go into. Why are there two files, and why is having a "models" package mandatory? If I try to deviate from what's on the website even a little bit, it seems the models file just doesn't ever get written.

**Comment by tv42** [comment]
Nothing gets written because

```
	if len(modelsBuild.Models) > 0 || len(modelsBuild.Enums) > 0 {
```

is false.

**Comment by tv42** [comment]
Ahhh models.gen.go only gets written for things that are inputs to mutations, it seems. Adding a mutation. That gets me forward. Okay, here's the thing I originally started chasing down: gqlgen rewrites dashes as underscores.

```
tv@dark ~/z/src/example.com/foo-bar$ ls -a
.gqlgen.yml  main.go  schema.graphql
tv@dark ~/z/src/example.com/foo-bar$ head -99 .gqlgen.yml main.go schema.graphql 
==> .gqlgen.yml <==
schema: schema.graphql
exec:
  package: main
model:
  package: example.com/foo-bar
models:
  Greeting:
    model: example.com/foo-bar.Greeting

==> main.go <==
package main

//go:generate gqlgen -package main -modelpackage graph -schema schema.graphql

type Greeting struct {
}

func main() {
}

==> schema.graphql <==
type Greeting {
  Message: String!
}

type Query {
  greeting: [Greeting!]!
}

input Hi {
  text: String!
}

type Mutation {
  sayHi(input: Hi!): Greeting!
}
tv@dark ~/z/src/example.com/foo-bar$ GOPATH="$HOME/z:$(go env GOPATH)" ~/go/bin/gqlgen 
panic: cannot find package "example.com/foo_bar" in any of:
	/home/tv/src/go-1.10/src/example.com/foo_bar (from $GOROOT)
	/home/tv/z/src/example.com/foo_bar (from $GOPATH)
	/home/tv/go/src/example.com/foo_bar

goroutine 1 [running]:
github.com/vektah/gqlgen/codegen.(*Imports).add(0xc420c41bc0, 0xc42429a860, 0x13, 0xc421300ed0)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/import_build.go:61 +0x2a3
github.com/vektah/gqlgen/codegen.buildImports(0xc427601b00, 0xc42013e330, 0x22, 0xc4275fb880)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/import_build.go:40 +0x194
github.com/vektah/gqlgen/codegen.(*Config).bind(0xc4201fbe18, 0xc42006f080, 0xc4201119b6, 0x2)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/build.go:66 +0xb6
github.com/vektah/gqlgen/codegen.Generate(0xc420111810, 0xe, 0xc420105680, 0x9e, 0xc42013e330, 0x2f, 0xc420111970, 0x4, 0xc42013e2d0, 0x30, ...)
	/home/tv/go/src/github.com/vektah/gqlgen/codegen/codegen.go:61 +0x5f1
main.main()
	/home/tv/go/src/github.com/vektah/gqlgen/main.go:81 +0x23d
```

**Comment by mathewbyrne** [comment]
@tv42 thanks for the report.  It looks like you've simultaneously run into a couple of issues that have been resolved in #179 and #180.

> If I try to deviate from what's on the website even a little bit, it seems the models file just doesn't ever get written.

In your first example, you had a single GraphQL type that was mapped to an existing struct.  In this case no model file is generated because there are no models needing generation.  If you were to add a second type to your schema and not map it, it would be generated for you.

> Ahhh models.gen.go only gets written for things that are inputs to mutations, it seems. Adding a mutation. 

Not strictly true, in your second example you now have an additional type `Hi` that does need to be generated.

The other problem with your second example is that you've misconfigured the model package.  This configuration option exists to set the name of the package at the top of the generated file.  In your case, omitting the `exec` and `model` config options, and relying on defaults should now work as expected with the latest release.
